src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -551,14 +551,14 @@ else if (conf.repair_session_space.toMebibytes() > (int) (Runtime.getRuntime().m
             conf.hints_directory = storagedirFor("hints");
         }
 
-        if (conf.native_transport_max_concurrent_requests_in_bytes <= 0)
+        if (conf.native_transport_max_request_data_in_flight == null || conf.native_transport_max_request_data_in_flight.toBytes() == 0)
```

Comment:
@maedhroz this is correct because the original check <=0 leads to the same result. In that sense I think set it to 0 was fine, but I agree it is more clear to use null, same as the permissions parameters. Though there using 0 was a real issue.  //cc @dcapwell